aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -191,9 +179,39 @@ def failure_handler(self, record, exception: ExceptionInfo) -> FailureResponse:
         return entry
 
 
+class SyncBasePartialProcessor(BasePartialProcessor):
+    @abstractmethod
+    def _process_record(self, record: dict):
+        """
+        Process record with handler.
+        """
+        raise NotImplementedError()
+
+    def process(self) -> List[Tuple]:
+        """
+        Call instance's handler for each record.
+        """
+        return [self._process_record(record) for record in self.records]
+
+
+class AsyncBasePartialProcessor(BasePartialProcessor):
+    @abstractmethod
+    async def _async_process_record(self, record: dict):
+        """
+        Process record with handler.
+        """
+        raise NotImplementedError()
+
+    async def async_process(self) -> List[Tuple]:
+        """
+        Call instance's handler for each record.
+        """
+        return list(await asyncio.gather(*[self._async_process_record(record) for record in self.records]))
```

Comment:
comment only: This makes me think we should also have a `concurrent=True` in the synchronous version, where we use a Future and document its caveats on thread-safe boto3 stuff.